syft/format/internal/cyclonedxutil/helpers/decoder.go

```diff
@@ -39,11 +39,32 @@ func ToSyftModel(bom *cyclonedx.BOM) (*sbom.SBOM, error) {
 }
 
 func collectBomPackages(bom *cyclonedx.BOM, s *sbom.SBOM, idMap map[string]interface{}) error {
-	if bom.Components == nil {
+	components := []cyclonedx.Component{}
```

Comment:
I think this change could be simplified to just adding this block:
```
	if bom.Metadata.Component != nil
		collectPackages(bom.Metadata.Component, s, idMap)
	}
```

... and in the `collectPackages`, we should be recursively adding sub-components, and checking for the type being one of the right types, [which it looks like it's doing already](https://github.com/anchore/syft/blob/9d40d1152e91fd520602f159e5d02a8736cb16ac/syft/format/internal/cyclonedxutil/helpers/decoder.go#L55)

Reply:
Sorry I should have looked at exactly what the code was doing there before sending you down a path to implement an unnecessary thing 🤦

Reply:
No worries, I also didn't see the logic was already present. Good thing you spotted it, I refactored it, the change is much simpler now :+1:

Reply:
I may have missed suggesting a `bom.Metadata != nil` bit

Reply:
Ah right, fixed